docs/src/modules/components/MarkdownDocsV2.js

```diff
@@ -0,0 +1,256 @@
+      hash: hookNameKebabCase,
+      children: hookToc,
+    });
+  });
+
+  function createComponentTocEntry(componentName, sectionName, hasInheritance = false) {
+    return {
+      text: getComponentTranslatedHeader(t, sectionName),
+      hash: `${componentName}-${sectionName}`,
+      children: [
+        ...(hasInheritance
+          ? [{ text: t('api-docs.inheritance'), hash: 'inheritance', children: [] }]
+          : []),
+      ],
+    };
+  }
+
+  const componentsApiToc = [];
+
+  Object.keys(componentsApiPageContents).forEach((key) => {
+    const { componentDescriptionToc = [] } = componentsApiDescriptions[key][userLanguage];
+
+    const { name: componentName } = componentsApiPageContents[key];
+
+    const componentNameKebabCase = kebabCase(componentName);
+
+    const componentApiToc = [
+      createComponentTocEntry(componentNameKebabCase, 'import'),
+      ...componentDescriptionToc,
+      componentsApiPageContents[key].styles.name &&
+        createComponentTocEntry(componentNameKebabCase, 'component-name'),
+      createComponentTocEntry(
+        componentNameKebabCase,
+        'props',
+        componentsApiPageContents[key].inheritance,
+      ),
+      componentsApiPageContents[key].styles.classes.length > 0 &&
+        createComponentTocEntry(componentNameKebabCase, 'css'),
+    ].filter(Boolean);
+
+    componentsApiToc.push({
+      text: componentName,
+      hash: componentNameKebabCase,
+      children: componentApiToc,
+    });
+  });
+
+  const isJoy = canonicalAs.startsWith('/joy-ui/');
+  const Provider = isJoy ? CssVarsProvider : React.Fragment;
+  const Wrapper = isJoy ? BrandingProvider : React.Fragment;
+
+  const commonElements = [];
+
+  let i = 0;
+  let done = false;
+
+  const wrapperProps = {
+    ...(isJoy && { mode: theme.palette.mode }),
+  };
+
+  // process the elements before the tabs component
+  while (i < rendered.length && !done) {
+    const renderedMarkdownOrDemo = rendered[i];
+    if (renderedMarkdownOrDemo.component && renderedMarkdownOrDemo.component.indexOf('Tabs') >= 0) {
+      done = true;
+    }
+    commonElements.push(
+      <MarkdownElement
+        srcComponents={srcComponents}
+        renderedMarkdownOrDemo={renderedMarkdownOrDemo}
+        WrapperComponent={Wrapper}
+        key={`common-elements-${i}`}
+        wrapperProps={wrapperProps}
+        localizedDoc={localizedDoc}
+        demos={demos}
+        location={location}
+        theme={theme}
+        demoComponents={demoComponents}
+        activeTab={activeTab}
+        disableAd={disableAd}
+      />,
+    );
+    i += 1;
+  }
+
+  let activeToc = demosToc;
+
+  if (activeTab === 'hooks-api') {
+    activeToc = hooksToc;
+  }
+
+  if (activeTab === 'components-api') {
+    activeToc = componentsApiToc;
+  }
+
+  return (
+    <AppLayoutDocs
+      description={description}
+      disableAd={disableAd}
+      disableToc={disableToc}
+      location={location}
+      title={title}
+      toc={activeToc}
```

Comment:
https://deploy-preview-35938--material-ui.netlify.app/base/react-tabs/components-api/#tabs-unstyled highlights the wrong header:

<img width="155" alt="Screenshot 2023-03-31 at 18 56 43" src="https://user-images.githubusercontent.com/3165635/229183096-f98f196f-2cfa-4873-9ce5-001f2b914ba9.png">